x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -133,6 +141,22 @@ protected void computeScores(RankFeatureDoc[] featureDocs, ActionListener<float[
                 return;
             }
 
+            // Resolve chunking settings if missing
+            ChunkScorerConfig resolvedChunkScorerConfig = resolveChunkingSettings(r);
+            if (resolvedChunkScorerConfig == null) {
+                l.onFailure(new IllegalArgumentException("Failed to resolve chunking settings for chunk_rescorer"));
```

Comment:
This means that everyone using semantic reranking as-is without chunking rescoring would break!